### PR TITLE
[plugin-web-app] Unite asserting element logic in web ui context

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/storage/AssertingWebElementsStorage.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.List;
+
+import com.google.common.eventbus.Subscribe;
+
+import org.openqa.selenium.WebElement;
+import org.vividus.softassert.event.AssertionEvent;
+import org.vividus.testcontext.TestContext;
+
+public class AssertingWebElementsStorage implements IAssertingWebElementsStorage
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/storage/AssertingWebElementsStorage.java:27]: why can't it be a part of `..UiContext`?

File: vividus-plugin-web-app/src/main/java/org/vividus/selenium/screenshot/ScreenshotTaker.java [hunk 2]
```diff
@@ -200,7 +202,7 @@ private Path takeScreenshot(Supplier<Path> screenshotFilePathSupplier,
             {
                 screenshotData = takeScreenshotAsByteArrayImpl(webElements, viewportScreenshot);
             }
-            webElementHighlighter.clearAssertingWebElements();
+            assertingWebElementsStorage.clearAssertingWebElements();
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/selenium/screenshot/ScreenshotTaker.java:205]: Is it needed considering new event-based clean up?

File: vividus-soft-assert/src/main/java/org/vividus/softassert/event/AssertionEvent.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.softassert.event;
+
+public interface AssertionEvent
+{
+    // Event marker
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/event/AssertionEvent.java:21]: that's a bad practice, just subscribe to all events

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/storage/AssertingWebElementsStorage.java (hunk 1 above)
Comment (uarlouski) [reply vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/storage/AssertingWebElementsStorage.java]: discussed

Comment (uarlouski) [reply vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/storage/AssertingWebElementsStorage.java]: Discussed that the WebUiContext is responsible for saving elements within its context and there is no need for separate component

**Review by valfirst — APPROVED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/DropdownSteps.java [hunk 4]
```diff
@@ -46,7 +46,7 @@
 
     @Inject private IWebElementActions webElementActions;
     @Inject private IBaseValidations baseValidations;
-    @Inject private IHighlightingSoftAssert highlightingSoftAssert;
+    @Inject private IDescriptiveSoftAssert highlightingSoftAssert;
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/DropdownSteps.java:49]: rename the field here and everywhere

**Review by ikalinin1 — CHANGES_REQUESTED** [review]

**Review by ikalinin1 — APPROVED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/context/WebUiContext.java [hunk 5]
```diff
@@ -109,15 +110,12 @@ public void clear()
...
-    public void putAssertingWebElements(List<WebElement> elements)
+    public boolean withAssertingWebElements(List<WebElement> elements, BooleanSupplier asserter)
     {
         testContext.put(ASSERTING_ELEMENTS_KEY, elements);
-    }
-
-    @Override
-    public void clearAssertingWebElements()
-    {
+        boolean outcome = asserter.getAsBoolean();
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/ui/web/context/WebUiContext.java:116]: it makes sense to wrap it in `try-final`

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**